finality-aleph/src/network/split.rs

```diff
@@ -101,8 +108,16 @@ impl<LeftData: Data, RightData: Data, R: ReceiverComponent<Split<LeftData, Right
     async fn next(&mut self) -> Option<LeftData> {
         loop {
             tokio::select! {
-                data = self.translated_receiver.next() => return data,
-                _ = forward_or_wait(&self.receiver, &self.left_sender, &self.right_sender) => (),
+                data = self.translated_receiver.next() => {
```

Comment:
Eh, I would drop both the `trace`s here (and in the right receiver too ofc).

Reply:
Done

Reply:
Doesn't seem to be.